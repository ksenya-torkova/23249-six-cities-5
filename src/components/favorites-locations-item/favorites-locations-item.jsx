import {favoritesLocationsItemTypes} from "../../prop-types";
import FavoritesCardsList from "../favorites-cards-list/favorites-cards-list";
import React from "react";

const FavoritesLocationsItem = (props) => {
  const {city, offers} = props;

  return (
    <li className="favorites__locations-items">
      <div className="favorites__locations locations locations--current">
        <div className="locations__item">
          <a className="locations__item-link" href="#">
            <span>{city}</span>
          </a>
        </div>
      </div>
      <div className="favorites__places">
        <FavoritesCardsList
          city = {city}
          offers = {offers}
        />
      </div>
    </li>
  );
};

FavoritesLocationsItem.propTypes = favoritesLocationsItemTypes;

export default FavoritesLocationsItem;
