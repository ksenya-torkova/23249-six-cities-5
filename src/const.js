
const cities = [
  `Amsterdam`,
  `Brussels`,
  `Cologne`,
  `Dusseldorf`,
  `Hamburg`,
  `Paris`,
];

const City = {
  AMSTERDAM: `Amsterdam`,
  BRUSSELS: `Brussels`,
  COLOGNE: `Cologne`,
  DUSSELDORF: `Dusseldorf`,
  HAMBURG: `Hamburg`,
  PARIS: `Paris`,
};

const SotringType = {
  HIGH_TO_LOW: `Price: high to low`,
  LOW_TO_HIGH: `Price: low to high`,
  POPULAR: `Popular`,
  TOP_RATED: `Top rated first`,
};

export {cities, City, SotringType};
