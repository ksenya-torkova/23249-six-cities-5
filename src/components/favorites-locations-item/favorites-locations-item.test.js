import {FavoritesLocationsItem} from "./favorites-locations-item";
import {mockCards} from "./../../const";
import React from "react";
import renderer from "react-test-renderer";

it(`Should FavoritesLocationsItem render properly`, () => {
  const tree = renderer
    .create(
        <FavoritesLocationsItem
          city={`Amsterdam`}
          offers={mockCards}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
