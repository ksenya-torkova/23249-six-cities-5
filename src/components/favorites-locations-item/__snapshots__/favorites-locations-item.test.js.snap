// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should FavoritesItem render properly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Amsterdam
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="place-card favorites__card"
      id={101}
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="place-card__image-wrapper favorites__image-wrapper"
      >
        <a
          href="/offer/101"
          onClick={[Function]}
        >
          <img
            alt="Place image"
            className="place-card__image"
            height="200"
            src="img/apartment-02.jpg"
            width="260"
          />
        </a>
      </div>
      <div
        className="place-card__info favorites__card-info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              100
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button place-card__bookmark-button--active button"
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              In bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
        >
          <a
            href="/offer/101"
            onClick={[Function]}
          >
            Wood and stone place
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
  </div>
</li>
`;
